ci/scenarios/edge/reroute/main.py

```diff
@@ -68,43 +47,44 @@ def main(blocked):
-        # Currently adding it statically, do not see another option
-        # remote_api = OVSClient('10.100.187.31', 'admin', 'admin')  # This is a non apt-ee envir and should not be able to connect here
-        # try:
-        #     compute_client = SSHClient('10.100.187.31', username='root', password='rooter')
-        # except Exception as ex:
-        #     raise RuntimeError('Could not setup the compute client. Got {}'.format(str(ex)))
-        vpool = None
-        for vp in VPoolHelper.get_vpools():
-            if len(vp.storagedrivers) >= 2 and vp.configuration['dtl_mode'] == 'sync':
-                vpool = vp
-                break
-        assert vpool is not None, 'Not enough vPools to test. We need at least a vPool with 2 storagedrivers'
-        available_storagedrivers = [storagedriver for storagedriver in vpool.storagedrivers]
-        std_1 = available_storagedrivers.pop(random.randrange(len(available_storagedrivers)))
-        std_2 = available_storagedrivers.pop(random.randrange(len(available_storagedrivers)))
-        str_1 = std_1.storagerouter  # Will act as volumedriver node
-        str_2 = std_2.storagerouter  # Will act as volumedriver node
-        str_3 = [storagerouter for storagerouter in StoragerouterHelper.get_storagerouters() if storagerouter.guid not in [str_1.guid, str_2.guid]][0]  # Will act as compute node
-        compute_client = SSHClient(str_3, username='root')  # Compute node is excluded from all migrations and shutdowns
-        EdgeTester.LOGGER.info('Chosen destination storagedriver is: {0}'.format(std_1.storage_ip))
-        EdgeTester.LOGGER.info('Chosen original owning storagedriver is: {0}'.format(std_2.storage_ip))
+    @classmethod
+    def start_test(cls):
+        cluster_info, is_ee, fio_bin_loc = cls.setup()
+        cls.test_reroute_fio(fio_bin_loc, cluster_info, is_ee=is_ee)
+
+    @classmethod
+    def setup(cls, logger=LOGGER):
+        destination_str, source_str, compute_str = cls.get_storagerouters_for_ha()
+        destination_storagedriver = None
+        source_storagedriver = None
+        storagedrivers_domain_sorted = DomainHelper.get_storagedrivers_in_same_domain(
+            domain_guid=source_str.regular_domains[0])
+        for storagedriver in storagedrivers_domain_sorted:
+            if len(storagedriver.vpool.storagedrivers) < 2:
+                continue
+            if storagedriver.guid in destination_str.storagedrivers_guids:
+                if destination_storagedriver is None and (
+                        source_storagedriver is None or source_storagedriver.vpool_guid == storagedriver.vpool_guid):
+                    destination_storagedriver = storagedriver
+                    logger.info('Chosen destination storagedriver is: {0}'.format(destination_storagedriver.storage_ip))
+                continue
+            if storagedriver.guid in source_str.storagedrivers_guids:
```

Comment:
Guids should be unique to an object.
However still a valid statement.
The second part of the check 
```
destination_storagedriver is None or destination_storagedriver.vpool_guid == storagedriver.vpool_guid
```
Should avoid the second concern
```